File: app/src/main/java/com/example/myappcontacts/di/contactsmap/ContactsMapComponent.java
package com.example.myappcontacts.di.contactsmap;

import com.example.myappcontacts.presentation.contactsmap.presenter.ContactsMapPresenter;

import dagger.Subcomponent;

@Subcomponent(modules = ContactsMapModule.class)
@ContactsMapScope
public interface ContactsMapComponent {
    void inject(ContactsMapPresenter contactsMapPresenter);
}

File: app/src/main/java/com/example/myappcontacts/di/contactsmap/ContactsMapComponent.kt
package com.example.myappcontacts.di.contactsmap

import com.example.myappcontacts.presentation.contactsmap.presenter.ContactsMapPresenter

import dagger.Subcomponent

@Subcomponent(modules = [ContactsMapModule::class])
@ContactsMapScope
interface ContactsMapComponent {

    fun inject(contactsMapPresenter: ContactsMapPresenter)
}
